Assets/Scripts/Models/Area/RoomManager.cs

```diff
@@ -0,0 +1,427 @@
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ProjectPorcupine.Rooms
+{
+    public class RoomManager : IEnumerable<Room>
+    {
+        /// <summary>
+        /// A all the currently managed rooms.
+        /// </summary>
+        private List<Room> rooms;
+
+        public RoomManager()
+        {
+            rooms = new List<Room>();
+
+            // Create the outside room.
+            OutsideRoom = new Room();
+
+            // Add the outside rooom to the list just in case.
+            rooms.Add(OutsideRoom);
+        }
+
+        #region Events
+
+        /// <summary>
+        /// Occurs when adding a new room to the manager, 
+        /// before the room is actually added.
+        /// </summary>
+        public event Action<Room> Adding;
+
+        /// <summary>
+        /// Occurs after the room has been added to the manager.
+        /// </summary>
+        public event Action<Room> Added;
+
+        /// <summary>
+        /// Occurs when a room is split.
+        /// The first room is the old one, 
+        /// the second is the new one.
+        /// </summary>
+        public event Action<Room, Room> Split;
+
+        /// <summary>
+        /// Occurs when two rooms are joined.
+        /// The first room is the old one, 
+        /// the second is the new one.
+        /// 
+        /// Warning, the old room has already been removed from
+        /// the manager, and most of its data can be considered
+        /// stale. This is a possible FIXME.
+        /// </summary>
+        public event Action<Room, Room> Joined;
+
+        /// <summary>
+        /// Occurs when removing a room from the manager, 
+        /// before the room is actually removed.
+        /// </summary>
+        public event Action<Room> Removing;
+
+        /// <summary>
+        /// Occurs after the room has been removed 
+        /// from the manager.
+        /// </summary>
+        public event Action<Room> Removed;
+
+        #endregion
+
+        /// <summary>
+        /// Gets the amount of rooms.
+        /// </summary>
+        /// <value>The number of rooms managed.</value>
+        public int Count
+        {
+            get
+            { 
+                return rooms.Count;
+            }
+        }
+
+        /// <summary>
+        /// Gets the outside room.
+        /// </summary>
+        /// <value>The outside room.</value>
+        public Room OutsideRoom { get; private set; }
+
+        /// <summary>
+        /// Gets a managed room from an ID.
+        /// </summary>
+        /// <param name="index">The ID of the room.</param>
+        public Room this [int index]
```

Comment:
Yes on mono it won't crash, trust me on this :) VS will die, it's a bug in Stylecop ;)